cmd/clusterctl/internal/util/objs.go

```diff
@@ -105,42 +120,87 @@ func FixImages(objs []unstructured.Unstructured, alterImageFunc func(image strin
 	// look for resources of kind Deployment and alter the image
 	for i := range objs {
 		o := &objs[i]
-		if o.GetKind() != deploymentKind {
-			continue
+		err := fixDeploymentImages(o, alterImageFunc)
+		if err != nil {
+			return nil, err
 		}
-
-		// Convert Unstructured into a typed object
-		d := &appsv1.Deployment{}
-		if err := scheme.Scheme.Convert(o, d, nil); err != nil {
+		err = fixDaemonSetImages(o, alterImageFunc)
+		if err != nil {
 			return nil, err
 		}
+		objs[i] = *o
+	}
+	return objs, nil
+}
 
-		// Alter the image
-		for j := range d.Spec.Template.Spec.Containers {
-			container := d.Spec.Template.Spec.Containers[j]
-			image, err := alterImageFunc(container.Image)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to fix image for container %s in deployment %s", container.Name, d.Name)
-			}
-			container.Image = image
-			d.Spec.Template.Spec.Containers[j] = container
-		}
+func fixDeploymentImages(o *unstructured.Unstructured, alterImageFunc func(image string) (string, error)) error {
+	if o.GetKind() != deploymentKind {
+		return nil
+	}
 
-		for j := range d.Spec.Template.Spec.InitContainers {
-			container := d.Spec.Template.Spec.InitContainers[j]
-			image, err := alterImageFunc(container.Image)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to fix image for init container %s in deployment %s", container.Name, d.Name)
-			}
-			container.Image = image
-			d.Spec.Template.Spec.InitContainers[j] = container
-		}
+	// Convert Unstructured into a typed object
+	d := &appsv1.Deployment{}
+	if err := scheme.Scheme.Convert(o, d, nil); err != nil {
+		return err
+	}
 
-		// Convert typed object back to Unstructured
-		if err := scheme.Scheme.Convert(d, o, nil); err != nil {
-			return nil, err
+	var err error
+	d.Spec.Template.Spec, err = fixPodSpecImages(d.Spec.Template.Spec, alterImageFunc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to fix containers in deployment %s", d.Name)
+	}
+
+	// Convert typed object back to Unstructured
+	if err := scheme.Scheme.Convert(d, o, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+func fixDaemonSetImages(o *unstructured.Unstructured, alterImageFunc func(image string) (string, error)) error {
+	if o.GetKind() != daemonSetKind {
+		return nil
+	}
+
+	// Convert Unstructured into a typed object
+	d := &appsv1.DaemonSet{}
+	if err := scheme.Scheme.Convert(o, d, nil); err != nil {
+		return err
+	}
+
+	var err error
+	d.Spec.Template.Spec, err = fixPodSpecImages(d.Spec.Template.Spec, alterImageFunc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to fix containers in deamonSet %s", d.Name)
+	}
+	// Convert typed object back to Unstructured
+	if err := scheme.Scheme.Convert(d, o, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+func fixPodSpecImages(podSpec v1.PodSpec, alterImageFunc func(image string) (string, error)) (v1.PodSpec, error) {
```

Comment:
Could we rename the `v1` package to `corev1`?

Reply:
ok